src/wallet/node_rpc_proxy.cpp

```diff
@@ -82,28 +82,49 @@ void NodeRPCProxy::invalidate()
   m_rpc_payment_seed_hash = crypto::null_hash;
   m_rpc_payment_next_seed_hash = crypto::null_hash;
   m_height_time = 0;
+  m_target_height_time = 0;
   m_rpc_payment_diff = 0;
   m_rpc_payment_credits_per_hash_found = 0;
   m_rpc_payment_height = 0;
   m_rpc_payment_cookie = 0;
+  m_daemon_hard_forks.clear();
 }
 
-boost::optional<std::string> NodeRPCProxy::get_rpc_version(uint32_t &rpc_version)
+boost::optional<std::string> NodeRPCProxy::get_rpc_version(uint32_t &rpc_version, std::vector<std::pair<uint8_t, uint64_t>> &daemon_hard_forks, uint64_t &height, uint64_t &target_height)
 {
   if (m_offline)
     return boost::optional<std::string>("offline");
   if (m_rpc_version == 0)
   {
+    const time_t now = time(NULL);
     cryptonote::COMMAND_RPC_GET_VERSION::request req_t = AUTO_VAL_INIT(req_t);
     cryptonote::COMMAND_RPC_GET_VERSION::response resp_t = AUTO_VAL_INIT(resp_t);
     {
       const boost::lock_guard<boost::recursive_mutex> lock{m_daemon_rpc_mutex};
       bool r = net_utils::invoke_http_json_rpc("/json_rpc", "get_version", req_t, resp_t, m_http_client, rpc_timeout);
       RETURN_ON_RPC_RESPONSE_ERROR(r, epee::json_rpc::error{}, resp_t, "get_version");
     }
+
     m_rpc_version = resp_t.version;
+    for (const auto &hf : resp_t.hard_forks)
+      m_daemon_hard_forks.push_back(std::make_pair(hf.hf_version, hf.height));
```

Comment:
If this method is called multiple times on the same object, then it's `m_daemon_hard_forks` field will keep expanding indefinitely.

Reply:
Added `m_daemon_hard_forks.clear()` above the for loop for better defense